crates/driver/src/infra/api/error.rs

```diff
@@ -0,0 +1,104 @@
+use {
+    crate::{
+        domain::competition::{self, quote, solution},
+        infra::api,
+    },
+    hyper::StatusCode,
+    serde::Serialize,
+};
+
+#[derive(Debug, Clone, Copy, Serialize)]
+#[serde(rename_all = "PascalCase")]
+enum Type {
+    QuotingFailed,
+    SolverFailed,
+    SolutionNotFound,
+    DeadlineExceeded,
+    SimulationFailed,
+    Unknown,
+    TransactionPublishingFailed,
+    InvalidAuctionId,
+    MissingSurplusFee,
+    LiquidityError,
+}
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Error {
+    error_type: Type,
+    description: &'static str,
+}
+
+impl From<Type> for (StatusCode, axum::Json<Error>) {
+    fn from(value: Type) -> Self {
+        let (description, status_code) = match value {
+            Type::QuotingFailed => ("No valid quote found", StatusCode::BAD_REQUEST),
+            Type::SolverFailed => (
+                "Solver engine returned an invalid response",
+                StatusCode::BAD_REQUEST,
```

Comment:
This doesn't seem like a client error (so it should not be a 4xx error code). I think the most appropriate error would be a 502:

> 502 Bad Gateway
>    The server was acting as a gateway or proxy and received an invalid response from the upstream server.

Rationale being the driver is a "gateway" to a solver engine.